Add documentation for kopia repository server controller workflow (#2077)
kanisterio/kanister: docs/templates.rst

    Secret      ObjectReference
  }

RepositoryServers
-----------------

> RepositoryServers are a Kanister CustomResource that contain information about storage,
> repository, and server details which are required for running a Kopia repository server
> instance, which can be further used for data operation artifacts.

Reviewer: ```suggestion
RepositoryServers are a Kanister CustomResource that contain information about storage,
repository, and server details. These details are required to run a Kopia repository server
instance, which can then be used for data operation artifacts.
```

In contrast to Secrets and ConfigMaps, an ActionSet can optionally reference only a
single repository server, eliminating the need to name the RepositoryServer in the
Blueprint.

> The following examples should be helpful.

Reviewer: ```suggestion
The following examples should be helpful:
```
